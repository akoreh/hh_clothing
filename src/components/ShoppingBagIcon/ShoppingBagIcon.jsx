import React from 'react';
import { connect } from 'react-redux';

import { toggleCartDropdownHidden } from '../../redux/cart/cart.actions';

import {ShoppingBagIconContainer, ShoppingBagIconAsset, ItemCountContainer} from './ShoppingBagIcon.styles';

const ShoppingBagIcon = ({toggleCartDropdownHidden, itemCount}) => (
    <ShoppingBagIconContainer onClick={toggleCartDropdownHidden}>
        <ShoppingBagIconAsset />
        <ItemCountContainer>{itemCount}</ItemCountContainer>
    </ShoppingBagIconContainer>
);

const mapDispatchToProps = dispatch => ({
    toggleCartDropdownHidden: () => dispatch(toggleCartDropdownHidden())
});

export default connect(null, mapDispatchToProps)(ShoppingBagIcon);