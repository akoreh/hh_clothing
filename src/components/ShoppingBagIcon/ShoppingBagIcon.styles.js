import styled from 'styled-components';

import { ReactComponent as ShoppingBagIconSVG } from '../../assets/shopping-bag.svg';

export const ShoppingBagIconContainer = styled.div`
  width: 3rem;
  height: 3rem;
  position: relative;
  display: flex;
  align-items: center;
  justify-content: center;
  cursor: pointer;
`;

ShoppingBagIconContainer.displayName = 'ShoppingBagIconContainer';

export const ShoppingBagIconAsset = styled(ShoppingBagIconSVG)`
  width: 2rem;
  height: 2rem;
`;

export const ItemCountContainer = styled.span`
  position: absolute;
  font-size: 12px;
  font-weight: bold;
  bottom: 10px;
`;

ItemCountContainer.displayName = 'ItemCountContainer';
