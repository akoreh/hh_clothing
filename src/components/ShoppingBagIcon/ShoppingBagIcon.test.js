import React from 'react';
import { shallow } from 'enzyme';

import ShoppingBagIcon from './ShoppingBagIcon';

describe('ShoppingBagIcon component', () => {
    let wrapper;

    beforeEach(() => {
        const mockProps = {
            itemCount: 0
        };

        wrapper = shallow(<ShoppingBagIcon {...mockProps}/>);
    });

    it('should render ShoppingBagIcon', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should render the itemCount', () => {
        const itemCountRef = wrapper.find('ItemCountContainer');

        expect(itemCountRef.text()).toBe('0');
    });
});