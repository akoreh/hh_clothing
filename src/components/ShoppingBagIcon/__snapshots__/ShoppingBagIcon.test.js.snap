// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShoppingBagIcon component should render ShoppingBagIcon 1`] = `ShallowWrapper {}`;
